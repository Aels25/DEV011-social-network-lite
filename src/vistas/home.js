// home.js

function home(navigateTo) {
  const section = document.createElement('section');
  const title = document.createElement('h2');
  const buttonLogin = document.createElement('button');
  const buttonSignUp = document.createElement('button');
  const buttonGoogleLogin = document.createElement('button');

  buttonLogin.id = 'btnLogin';
  buttonSignUp.id = 'btnSignUp';
  buttonGoogleLogin.id = 'btnGoogleLogin'; // Asigna un ID único para el botón de inicio de sesión con Google

  title.textContent = 'BIENVENIDO A LA WANDERWEB';

  buttonLogin.textContent = 'Iniciar sesión';
  buttonLogin.addEventListener('click', () => {
    navigateTo('/login');
  });

  buttonSignUp.textContent = 'Crear cuenta nueva';
  buttonSignUp.addEventListener('click', () => {
    navigateTo('/signup');
  });

  buttonGoogleLogin.textContent = 'Iniciar sesión con Google';
  // Agrega un manejador de eventos al nuevo botón de Google
  buttonGoogleLogin.addEventListener('click', () => {
    navigateTo('/googlelogin'); // Redirige a la nueva vista de Google Login
  });

  section.append(title, buttonLogin, buttonSignUp, buttonGoogleLogin);
  return section;
}

export default home;
