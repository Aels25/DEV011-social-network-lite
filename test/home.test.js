/**
 * @jest-environment jsdom
 */

import home from '../src/vistas/home.js'; // Ajusta la ruta segÃºn tu estructura de archivos

describe('home', () => {
  test('snapshot of home', () => {
    const DOM = document.createElement('div');
    DOM.append(home(() => {}));
    expect(DOM).toMatchSnapshot();
  });

  test('is a function', () => {
    expect(typeof home).toBe('function');
  });

  test('have a button Login', () => {
    const DOM = document.createElement('div');
    DOM.append(home(() => {}));
    const haveAButton = DOM.querySelector('#btnLogin');
    expect(haveAButton).not.toBe(null);
  });

  test('have a button SignUp', () => {
    const DOM = document.createElement('div');
    DOM.append(home(() => {}));
    const haveAButton = DOM.querySelector('#btnSignUp');
    expect(haveAButton).not.toBe(null);
  });

  test('have a button GoogleLogin', () => {
    const DOM = document.createElement('div');
    DOM.append(home(() => {}));
    const haveAButton = DOM.querySelector('#btnGoogleLogin');
    expect(haveAButton).not.toBe(null);
  });

  test('button Login calls navigateTo on click', () => {
    const navigateTo = jest.fn();
    const DOM = document.createElement('div');
    DOM.append(home(navigateTo));
    const buttonLogin = DOM.querySelector('#btnLogin');
    buttonLogin.click();
    expect(navigateTo).toHaveBeenCalledTimes(1);
    expect(navigateTo).toHaveBeenCalledWith('/login');
  });

  test('button SignUp calls navigateTo on click', () => {
    const navigateTo = jest.fn();
    const DOM = document.createElement('div');
    DOM.append(home(navigateTo));
    const buttonSignUp = DOM.querySelector('#btnSignUp');
    buttonSignUp.click();
    expect(navigateTo).toHaveBeenCalledTimes(1);
    expect(navigateTo).toHaveBeenCalledWith('/signup');
  });

  test('button GoogleLogin calls navigateTo on click', () => {
    const navigateTo = jest.fn();
    const DOM = document.createElement('div');
    DOM.append(home(navigateTo));
    const buttonGoogleLogin = DOM.querySelector('#btnGoogleLogin');
    buttonGoogleLogin.click();
    expect(navigateTo).toHaveBeenCalledTimes(1);
    expect(navigateTo).toHaveBeenCalledWith('/googlelogin');
  });
});
