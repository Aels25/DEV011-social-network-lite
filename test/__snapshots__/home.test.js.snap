// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home snapshot of home 1`] = `
<div>
  <section>
    <h2>
      BIENVENIDO A LA WANDERWEB
    </h2>
    <button
      id="btnLogin"
    >
      Iniciar sesión
    </button>
    <button
      id="btnSignUp"
    >
      Crear cuenta nueva
    </button>
    <button
      id="btnGoogleLogin"
    >
      Iniciar sesión con Google
    </button>
  </section>
</div>
`;
